launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -336,4 +334,18 @@ static void addPermGenSizeOpt(List<String> cmd) {
     cmd.add("-XX:MaxPermSize=256m");
   }
 
+  /**
+   * Get the major version of the java.version string supplied.
+   */
+  static int javaMajorVersion(String javaVersion) {
+    String[] version = javaVersion.split("[+.\\-]+");
```

Comment:
oh yah sorry not enough coffee for me today, I guess we need .- but not + (although if we did use java.runtime.version  we would need the +) and one of the samples in the test "9+100" shouldn't show up in java.version but should in java.runtime.version